firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java

```diff
@@ -680,6 +682,85 @@ public void testEncodesMultipleFiltersOnDeeperCollections() {
         serializer.decodeQueryTarget(serializer.encodeQueryTarget(q.toTarget())), q.toTarget());
   }
 
+  @Test
+  public void testEncodesCompositeFiltersOnDeeperCollections() {
+    // (prop < 42) || (author == "dimond" && tags array-contains "pending")
+    Query q =
+        Query.atPath(ResourcePath.fromString("rooms/1/messages/10/attachments"))
+            .filter(
+                orFilter(
+                    filter("prop", "<", 42),
+                    andFilter(
+                        filter("author", "==", "dimond"),
+                        filter("tags", "array-contains", "pending"))));
```

Comment:
FWIW just a comment on the naming here. If we use my new favorite names anyOf/allOf then it will be clear right away that we are not anding `filter("prop", "<", 42)` with ` andFilter(filter("author", "==", "dimond"),...`):


```
  Query q =
        Query.atPath(ResourcePath.fromString("rooms/1/messages/10/attachments"))
            .filter(
                anyOf(
                    filter("prop", "<", 42),
                    allOf(
                        filter("author", "==", "dimond"),
                        filter("tags", "array-contains", "pending"))));

```

Please ignore :)